### PR TITLE
internal/mysql: Port tests to endtoend package

### PR DESCRIPTION
**Pull request by kyleconroy**
As part of my work on #302, I wanted to avoid having to update the MySQL tests to use the new configuration format. When tests are written using the end-to-end framework, none of the internal structures leak into the tests.

### CODE REVIEW

**Comment by kyleconroy** [comment]
@cmoog Care to take a look?

File: internal/endtoend/testdata/mysql_enums/go/models.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+package querytest
+
+import ()
+
+type FirstNameType string
+
+const (
+	john   FirstNameType = "john"
+	albert FirstNameType = "albert"
```
Comment (kyleconroy) [inline internal/endtoend/testdata/mysql_enums/go/models.go]: We're going to have to fix the naming here. These should be

```go
const (
	FirstNameJohn   FirstNameType = "john"
	FirstNameAlbert FirstNameType = "albert"
)
```

Comment (kyleconroy) [reply internal/endtoend/testdata/mysql_enums/go/models.go]: It's fine for now, will address it in another PR.

Comment (cmoog) [reply internal/endtoend/testdata/mysql_enums/go/models.go]: Good find. I must have written some unfortunately narrow tests that didn't catch that before.

**Comment by cmoog** [comment]
Looks good to me 👍

**Pull request merged**